tests/components/blebox/test_config_flow.py

```diff
@@ -0,0 +1,192 @@
+        assert result["errors"] == {"base": "cannot_connect"}
+
+
+async def test_flow_with_unknown_failure(hass, product_class_mock):
+    """Test that config flow works."""
+    with product_class_mock as products_class:
+        products_class.async_from_host = AsyncMock(side_effect=RuntimeError)
+        result = await hass.config_entries.flow.async_init(
+            config_flow.DOMAIN,
+            context={"source": config_entries.SOURCE_USER},
+            data={config_flow.CONF_HOST: "172.2.3.4", config_flow.CONF_PORT: 80},
+        )
+        assert result["errors"] == {"base": "unknown"}
+
+
+async def test_flow_with_unsupported_version(hass, product_class_mock):
+    """Test that config flow works."""
+    with product_class_mock as products_class:
+        products_class.async_from_host = AsyncMock(
+            side_effect=blebox_uniapi.error.UnsupportedBoxVersion
+        )
+
+        result = await hass.config_entries.flow.async_init(
+            config_flow.DOMAIN,
+            context={"source": config_entries.SOURCE_USER},
+            data={config_flow.CONF_HOST: "172.2.3.4", config_flow.CONF_PORT: 80},
+        )
+        assert result["errors"] == {"base": "unsupported_version"}
+
+
+async def test_async_setup(hass):
+    """Test async_setup (for coverage)."""
+    assert await async_setup_component(hass, "blebox", {"host": "172.2.3.4"})
+    await hass.async_block_till_done()
+
+
+async def test_already_configured(hass, valid_feature_mock):
+    """Test that same device cannot be added twice."""
+
+    config = mock_config("172.2.3.4")
+    config.add_to_hass(hass)
+
+    await hass.config_entries.async_setup(config.entry_id)
+    await hass.async_block_till_done()
+
+    result = await hass.config_entries.flow.async_init(
+        config_flow.DOMAIN,
+        context={"source": config_entries.SOURCE_USER},
+        data={config_flow.CONF_HOST: "172.2.3.4", config_flow.CONF_PORT: 80},
+    )
+    assert result["type"] == data_entry_flow.RESULT_TYPE_ABORT
+    assert result["reason"] == "address_already_configured"
+
+
+async def test_async_setup_entry(hass, valid_feature_mock):
+    """Test async_setup_entry (for coverage)."""
+
+    config = mock_config()
+    config.add_to_hass(hass)
+
+    assert await hass.config_entries.async_setup(config.entry_id)
+    await hass.async_block_till_done()
+
+    assert hass.config_entries.async_entries() == [config]
+    assert config.state == config_entries.ENTRY_STATE_LOADED
+
+
+async def test_async_remove_entry(hass, valid_feature_mock):
+    """Test async_setup_entry (for coverage)."""
+
+    config = mock_config()
+    config.add_to_hass(hass)
+
+    assert await hass.config_entries.async_setup(config.entry_id)
+    await hass.async_block_till_done()
+
+    assert await hass.config_entries.async_remove(config.entry_id)
+    await hass.async_block_till_done()
+
+    assert hass.config_entries.async_entries() == []
+    assert config.state == config_entries.ENTRY_STATE_NOT_LOADED
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5462445)